File: cinema_portal_web/src/main/webapp/js/service/cityService.js
//服务层
app.service('cityService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../city/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../city/findPage.do?pageNum='+page+'&pageSize='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../city/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../city/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../city/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../city/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../city/search.do?pageNum='+page+"&pageSize="+rows, searchEntity);
	}    	
});

File: cinema_portal_interface/src/main/java/cn/edu/aynu/rjxy/portal/service/CityService.java
package cn.edu.aynu.rjxy.portal.service;


import cn.edu.aynu.rjxy.pojo.TbCity;
import com.github.pagehelper.PageInfo;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface CityService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbCity> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageInfo findPage(int pageNum, int pageSize);


	/**
	 * 增加
	*/
	public void add(TbCity city);


	/**
	 * 修改
	 */
	public void update(TbCity city);


	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbCity findOne(Long cityId);


	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] cityIds);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageInfo findPage(TbCity city, int pageNum, int pageSize);
	
}
